### ISSUE TITLE
 Another issue with git-annex detection

### ISSUE THREAD

**Issue by phil-conundrum** [G-Node/gin-cli#147]
Hello everyone!

When trying to log into the Gin system via the command line client it returns:
"PS C:> gin login
The GIN Client requires git-annex 6.20160126 or newer"

However, this required version of git-annex is already installed (6.20171205-ge5c610b29).

What do you guys think?

This might be related top [#121](https://github.com/G-Node/gin-cli/issues/121)

Versions used:
Gin: GIN command line client 0.12beta Build 000699 (83ec709f55f74f7c06be293acd19dd02b1fa4be1)
Git: git version 2.15.1.windows.2
Git-Annex: git annex version --raw 6.20171205-ge5c610b29PS

**Comment by Schopi** [comment]
[gin.log](https://github.com/G-Node/gin-cli/files/1615450/gin.log)

**Comment by achilleas-k** [comment]
Yes, this is definitely an instance of the same issue reported in #121.

The version you're using includes a fix for that particular version string, but it seems we need a more general solution. Thanks for the report. I'll see if I can get a new beta out with a fix for this today.

**Comment by achilleas-k** [comment]
Just out of curiosity, how did you install git-annex? It doesn't really matter—I'm still going to make the version check as general as possible—but I'm always curious to know where the version string variant came from.

**Comment by achilleas-k** [comment]
This is strange. I can't reproduce the error. In fact, the version string `6.20171205-ge5c610b29` works fine even with older versions of the client (v0.11), before the fix for #121 was added.

I wonder if gin is using a different git-annex binary than the one called on the PS command line. It shouldn't, but there might be some environment or PATH mixup.

Can you try running `gin annex version`? (note this is **gin** annex, not git)
This will run `git-annex version` using whichever git-annex binary the gin client uses.

**Comment by Schopi** [comment]
Regarding your first question on how I installed git-annex, I followed the basic git-annex installation guide for Windows (https://git-annex.branchable.com/install/Windows/) after I installed Git similarly. The referenced Git installation installs Git under C:\Users\<User>\AppData\Local\Programs and hence I pointed the git-annex installer to C:\Users\<User>\AppData\Local\Programs\Git. 

Regarding your second question, I will be at my computer only by next week Tuesday but here is the context of my problem: I tried to install the GinGUI local client for Windows (https://github.com/G-Node/GinUI), which asked me to "Add all the paths set by gin_shell.bat to the SYSTEM path."

**Comment by achilleas-k** [comment]
Interesting. Just to clarify, the most straightforward way to get gin-cli working on Windows is to use the portable bundle we provide (see the quick start page on the [gin wiki](https://web.gin.g-node.org/G-Node/Info/wiki/GinCli#windows)). The provided zip file contains portable versions of git, git-annex, and a batch file that sets the environment and starts a shell, so everything is ready to use.

However, I would still be interested to know what kind of issues arise when users follow the installation procedure you describe. For instance, users/developers with an existing, systemwide git and git annex installation may not want to have a separate installation of those tools.

Now as for the GinUI, the installation instructions also assume that git and git annex are available as they are bundled through gin. However, it still should (theoretically) work if paths etc are set correctly.

Let me know what the output of `gin annex version` is when you get the chance and we can figure out what's going wrong.

**Comment by Schopi** [comment]
Good Morning!

Back in the office now. 
Running `gin annex version` from the systemwide powershell yields:

```
PS C:\Users\<User>> gin annex version
git-annex version: 6.20171205-ge5c610b29
build flags: Assistant Webapp Pairing Testsuite S3(multipartupload)(storageclasses) WebDAV ConcurrentOutput TorrentParser Feeds
dependency versions: aws-0.17.1 bloomfilter-2.0.1.0 cryptonite-0.23 DAV-1.3.1 feed-0.3.12.0 ghc-8.0.2 http-client-0.5.7.1 persistent-sqlite-2.6.2 torrent-10000.1.1 uuid-1.3.13 yesod-1.4.5
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384 SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_224 SHA3_384E SHA3_384 SKEIN256E SKEIN256 SKEIN512E SKEIN512 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav tahoe glacier ddar hook external
```

Running `gin annex version` from the G-Node shipped Gin client yields:

```
C:\Users\<User>>gin annex version
git-annex version: 6.20171205-ge5c610b29
build flags: Assistant Webapp Pairing Testsuite S3(multipartupload)(storageclasses) WebDAV ConcurrentOutput TorrentParser Feeds
dependency versions: aws-0.17.1 bloomfilter-2.0.1.0 cryptonite-0.23 DAV-1.3.1 fe
ed-0.3.12.0 ghc-8.0.2 http-client-0.5.7.1 persistent-sqlite-2.6.2 torrent-10000.1.1 uuid-1.3.13 yesod-1.4.5
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384
SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_224 SHA3_384E SHA3_384 SKEI
N256E SKEIN256 SKEIN512E SKEIN512 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav tahoeglacier ddar hook external
```

**Comment by achilleas-k** [comment]
This is very strange. I'll follow your installation steps to reproduce this.

Thanks for the output.

**Comment by Schopi** [comment]
This is for the **situation with both** Gins shell.bat paths set as systemwide paths and **with** previous git or git-annex paths. Here the error occurs.
The Paths are:
`C:\Users\Louis\AppData\Local\Programs\Git\cmd;C:\gin\bin;C:\gin\git\usr\bin;C:\gin\git\bin`
Running gin annex version from the systemwide cmd yields:
```
C:\Users\Louis>gin annex version
The GIN Client requires git-annex 6.20160126 or newer
git-annex version
C:\Users\Louis>git annex version --raw
6.20171205-ge5c610b29 not understood
```

Running gin annex version from the shipped GinClient yields: 
```
C:\Users\Louis>gin annex version
git-annex version: 6.20171205-ge5c610b29
build flags: Assistant Webapp Pairing Testsuite S3(multipartupload)(storageclass
es) WebDAV ConcurrentOutput TorrentParser Feeds
dependency versions: aws-0.17.1 bloomfilter-2.0.1.0 cryptonite-0.23 DAV-1.3.1 fe
ed-0.3.12.0 ghc-8.0.2 http-client-0.5.7.1 persistent-sqlite-2.6.2 torrent-10000.
1.1 uuid-1.3.13 yesod-1.4.5
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384
SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_224 SHA3_384E SHA3_384 SKEI
N256E SKEIN256 SKEIN512E SKEIN512 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav tahoe
glacier ddar hook external
```

**Comment by Schopi** [comment]
Sorry for the confusion. I did not realise that I changed the paths before leaving the office last week. I tried to distinguish both situations above. I hope it helps.

**Comment by achilleas-k** [comment]
Did something go wrong with the copy-paste below?
```
C:\Users\Louis>gin annex version
The GIN Client requires git-annex 6.20160126 or newer
git-annex version
C:\Users\Louis>git annex version --raw
6.20171205-ge5c610b29 not understood
```

If this is how you got the output, it might be an issue with new lines or carriage returns in the version string.

**Comment by achilleas-k** [comment]
Fixed in #241.
